src/interface/block_chain.cpp

```diff
@@ -1536,7 +1536,7 @@ void block_chain::fetch_history(const hash_digest& script_hash, size_t limit,
     for (auto payment: database_.addresses().get(script_hash))
     {
         // The limit is not so useful due to lack of ordering.
-        if (count++ == limit)
+        if ((limit != 0) && (count++ == limit))
```

Comment:
@thecodefactory This appears to be a bug, as the counter will not update unless `limit == 0`.

Reply:
This may be a bug, but it's not clear to me yet.  The idea is that if `limit == 0`, we don't consider the count, as there is no limit. If limit is anything other than 0, we consider the count and the specified limitation.

Reply:
yes, I see. The count is only used within the condition. That’s fine. For some reason I imagined it having a use outside of the condition (eg reporting the actual count). I’m always weary of multipart conditions where one part contains a modifier.